src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -996,17 +996,27 @@ static bool patternMatchesNull(BoundPattern pattern)
                     case BoundNegatedPattern negated:
                         return !patternMatchesNull(negated.Negated);
                     case BoundBinaryPattern binary:
-                        if (binary.Disjunction)
+                        var binaryPatterns = getBinaryPatterns(binary);
```

Comment:
>var binaryPatterns = getBinaryPatterns(binary); [](http://example.com/codeflow?start=24&length=47)

I would expect to see change to an override of VisitBinaryPattern in this class. And, if we don't expect it to be called, throw from it.